src/main/kotlin/io.javalin/builder/CookieBuilder.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Javalin - https://javalin.io
+ * Copyright 2017 David Åse
+ * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
+ */
+
+package io.javalin.builder
+
+import io.javalin.core.util.Util
+
+//TODO: This auto-conversion could probably be cleaned up a lot
+class CookieBuilder private constructor(private val name: String, private val value: String) {
+
+    private var domain = ""
+    private var path = ""
+    private var maxAge = -1
+    private var secure = false
+    private var httpOnly = false
+
+    companion object {
+        fun cookieBuilder(name: String, value: String): CookieBuilder {
+            return CookieBuilder(name, value)
+        }
+    }
+
+    init {
+        Util.notNull(name, "Cookie name cannot be null")
+        Util.notNull(value, "Cookie value cannot be null")
+    }
+
+    fun domain(domain: String): CookieBuilder {
+        this.domain = domain
```

Comment:
maybe you could throw an exception if domain is already set. The same goes for the other parameters